lib/services/local/access_monitoring_rules.go

```diff
@@ -34,7 +35,8 @@ const accessMonitoringRulesPrefix = "access_monitoring_rule"
 
 // AccessMonitoringRulesService manages AccessMonitoringRules in the Backend.
 type AccessMonitoringRulesService struct {
-	svc *generic.ServiceWrapper[*accessmonitoringrulesv1.AccessMonitoringRule]
+	backend backend.Backend
```

Comment:
instead of passing `backend.Backend`, can't you handle the list method instead?

a service with two backends is confusing.

Reply:
I am unsure of what you mean here. The list with filter is handled here and the normal list is handled by the generic service wrapper.

Reply:
@tigrato  could you clarify what is  you meant in this comment?

Reply:
Essencially `*generic.ServiceWrapper[*accessmonitoringrulesv1.AccessMonitoringRule]` should abstract the logic from using the backend itself. 

Adding the backend here to implement the filter - adding some overlap logic with `generic.ServiceWrapper` - is a bit confusing.

I would prefer to add a ListResourcesWithFilter method to `*generic.ServiceWrapper` and `generic.Service`  where you can pass a matcher function to it and all the logic is abstracted. This would simplify any other future work

Reply:
Something similar to this https://github.com/gravitational/teleport/commit/e5b02b8e247ecf6b7477697fe17db2a216a40393

I didn't test this code and wrote it without thinking too much

Reply:
Retested it and your suggestion seems to work out of the box.